### ISSUE TITLE
Cancelled member who signs up via Stripe recurring incorrectly receives "Expired" email

### ISSUE THREAD

**Issue by nosegraze** [restrictcontentpro/restrict-content-pro#1078]
To reproduce:

- Start with a "Cancelled" member who hasn't yet reached their expiration date.
- Manually change the expiration date in the admin area to be in the past.
- Register for a new _recurring_ subscription via Stripe.
- Immediately after payment, visit the "Your Membership" page ( [subscription_details] shortcode).

The member should receive the "Your account has expired" email, followed by membership activation email.

This happens because immediately after the Stripe payment, we change the status to 'active' but we DO NOT update the expiration date (not for recurring members). Then if you visit the "Your Membership" page before the webhook kicks in, RCP displays the membership status, which uses `RCP_Member::get_status()`, which has this little snippet:

```php
// double check that the status and expiration match. Update if needed
if( $status == 'active' && $this->is_expired() ) {

	$status = 'expired';
	$this->set_status( $status );

}
```

Before the webhook is triggered, the member's status is set to 'active' but the expiration date is still in the past, so the status is quickly set to 'expired', the email is sent, then the webhook kicks in and switches the status back to 'active'.

**Comment by nosegraze** [comment]
PR: #1079

**Comment by pippinsplugins** [comment]
I may be remembering wrong, but I believe we had it setup to put recurring members immediately to active because it mitigated occasional problems due to delayed webhooks and customers not seeing their account as being active after completing the payment.

As long as we all recognize that is a potential side effect here, I'm fine with this change.

**Comment by mindctrl** [comment]
Sounds like another issue that would be solved by marking cancelled users as expired when they reach the expiration date. I'm fine with this change in the interim.

**Comment by nosegraze** [comment]
What about this: https://github.com/restrictcontentpro/restrict-content-pro/commit/23bd5c550e92a52c3373c70a07c758941e42dffd

With that change, a brand new user will see "Active" immediately. A cancelled user with an expired account will see "Pending" for a few seconds. This is actually better than what currently happens, which is that they'd see "Expired". :P

**Comment by pippinsplugins** [comment]
Yes . . . I think that's good.

**Comment by thomasfw** [comment]
I contacted support about this a while ago but didn't get anywhere with a fix, so i'm glad this has been picked up. For the past year all of our lapsed members have been receiving the 'Expired' email after renewing their membership, and at the same time as the 'Active' email (just had a search and it also sounds like #241). 

Looks like these two issues are probably related, and the proposed fix will hopefully put a stop to this too?

```
July 2, 2016 01:39:26 - The subscription was set up with the batch user import.

February 1, 2017 17:20:47 - Member's status changed from active to expired

/****** User renews here ******/
February 13, 2017 10:30:32 - Member's status changed from expired to pending

February 13, 2017 10:30:38 - Member's status changed from pending to active

February 13, 2017 10:30:40 - Member's status changed from active to expired

February 13, 2017 10:30:44 - Member's expiration changed from 2017-01-31 23:59:59 to 2018-02-13 23:59:59

February 13, 2017 10:30:44 - Member's status changed from expired to active
```

**Comment by nosegraze** [comment]
@thomasfw Are your customers also using Stripe?

**Comment by mindctrl** [comment]
Stripe also responds with the `current_period_end1` date when creating a new subscription. We could set the expiration date during the initial subscription creation instead of waiting on the webhook.

**Comment by thomasfw** [comment]
@nosegraze yep we're using Stripe

**Comment by mindctrl** [comment]
I updated this to set the expiration date during the initial signup for auto-renewing subscriptions. I can't think of a reason to do this, and it matches the expiration date they send seconds/minutes later in the webhook.

**Comment by nosegraze** [comment]
This looks good to me! Gave it a test and it worked fine.
